pkg/model/planpreview.proto

```diff
@@ -44,15 +44,13 @@ message ApplicationPlanPreviewResult {
     ApplicationKind application_kind = 4 [(validate.rules).enum.defined_only = true];
     string application_directory = 5 [(validate.rules).string.min_len = 1];
 
-    string env_id = 6;
-    string env_name = 7;
-    // Web URL to the environment page.
-    // This is only filled before returning to the client.
-    string env_url = 8;
+    reserved 6, 7, 8;
```

Comment:
Reserving only the index number but not the name can cause some issues for JSON serialization.
Even though this ApplicationPlanPreviewResult is not being stored in the data store currently but anything inside this `model` package can be encoded by JSON before storing or passing around.

Therefore, we have to reserve both name and index number.

But, Protocol Buffer does not allow mixing field names and field numbers in the same reserved statement. So I propose to use `deprecated` instead.

```
string env_id = 6 [deprecated = true];
string env_name = 7 [deprecated = true];
string env_url = 8 [deprecated = true];
```

Reply:
Thank you!
Is it wrong to reserve names for another reserved statement?

Reply:
> Is it wrong to reserve names for another reserved statement?

No, that way is ok too.
Just because we are using `deprecated` at other places in that package so I propose to use it to get unified.